internal/controller/isbservicerollout/isbservicerollout_controller.go

```diff
@@ -204,6 +204,11 @@ func (r *ISBServiceRolloutReconciler) reconcile(ctx context.Context, isbServiceR
 		numaLogger.Info("Deleting ISBServiceRollout")
 		if controllerutil.ContainsFinalizer(isbServiceRollout, common.FinalizerName) {
 			ppnd.GetPauseModule().DeletePauseRequest(isbsvcKey)
+			// Set the foreground deletion policy
+			foreground := metav1.DeletePropagationForeground
+			if err := r.client.Delete(ctx, isbServiceRollout, &client.DeleteOptions{PropagationPolicy: &foreground}); err != nil {
+				return ctrl.Result{}, err
+			}
```

Comment:
This is really interesting. So, from [this](https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/) article, it looks like doing this essentially creates a finalizer. 

```
"finalizers": [
      "foregroundDeletion"
    ]
```

And then the GarbageCollector Controller is actually in charge of removing that finalizer. 

Good find!

Reply:
Yes